# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../shadow/shadow-4.1.5-stdarg.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From 04d19a920ea0cf74f977006e3f8241bf77cbc16b Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Sun, 12 Feb 2012 22:41:05 -0500
Subject: [PATCH] copydir: include stdarg.h for va_* funcs

Building on newer systems fails due to using stdargs but not including
the header file:

copydir.c: In function 'error_acl':
copydir.c:126:2: warning: implicit declaration of function 'va_start' [-Wimplicit-function-declaration]
copydir.c:132:2: warning: implicit declaration of function 'va_end' [-Wimplicit-function-declaration]
...
libmisc/copydir.c:126: undefined reference to 'va_start'
libmisc/copydir.c:132: undefined reference to 'va_end'
...
make[2]: *** [useradd] Error 1

Signed-off-by: Mike Frysinger <vapier@gentoo.org>
---
 libmisc/copydir.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/libmisc/copydir.c b/libmisc/copydir.c
index 7cb5f56..c6cbe6b 100644
--- a/libmisc/copydir.c
+++ b/libmisc/copydir.c
@@ -40,6 +40,7 @@
 #include <sys/time.h>
 #include <fcntl.h>
 #include <stdio.h>
+#include <stdarg.h>
 #include "prototypes.h"
 #include "defines.h"
 #ifdef WITH_SELINUX
-- 
1.7.8.4

